### PR TITLE
Fix UBSAN failure caused by left-shift of negative number

### LINKED ISSUE #1920: UBSAN failure due to left-shift of negative number in Exiv2::Internal::PentaxMakerNote::printDate
Found by OSS-Fuzz: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39060

To reproduce, you need a sanitizer build:

```bash
mkdir build-asan
build-asan/
cmake -DCMAKE_BUILD_TYPE=Debug -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DEXIV2_TEAM_USE_SANITIZERS=ON ..
make
./bin/exiv2 -Pt poc.tiff
```

poc: [poc.tar.gz](https://github.com/Exiv2/exiv2/files/7214222/poc.tar.gz)

Error message:

```
/home/kev/work/exiv2/src/pentaxmn_int.cpp:1039:33: runtime error: left shift of negative value -1
```

I do not consider this a security bug, because the left shift works perfectly fine in a normal build.

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Fixes: #1920 

Credit to OSS-Fuzz: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39060

Add a cast to `uint64_t` to fix a UBSAN failure.

### CODE REVIEW

**Comment by kevinbackhouse** [comment]
In the third commit, I added a CodeQL query which warns about other signed shifts in the codebase. It has generated quite a few warnings, which I plan to fix in a separate PR, so that the warnings show up as "closed" on the [Security](https://github.com/Exiv2/exiv2/security) tab.

File: src/pentaxmn_int.cpp [hunk 1]
```diff
@@ -1036,7 +1036,7 @@ namespace Exiv2 {
     std::ostream& PentaxMakerNote::printDate(std::ostream& os, const Value& value, const ExifData*)
     {
         /* I choose same format as is used inside EXIF itself */
-        os << ((value.toLong(0) << 8) + value.toLong(1));
+        os << ((static_cast<uint64_t>(value.toLong(0)) << 8) + value.toLong(1));
```
Comment (kmilos) [inline src/pentaxmn_int.cpp:1039]: uint32_t doesn't cut it here?

**Review by kmilos — COMMENTED** [review]
What about all the other warnings now?

**Comment by kevinbackhouse** [comment]
I can put the CodeQL query in a separate PR if you prefer. It has generated a lot of spam in the diff.

**Review by kmilos — APPROVED** [review]
LGTM

**Pull request merged**